coderd/database/dbmem/dbmem.go

```diff
@@ -3625,16 +3625,24 @@ func (q *FakeQuerier) GetProvisionerDaemons(_ context.Context) ([]database.Provi
 	return out, nil
 }
 
-func (q *FakeQuerier) GetProvisionerDaemonsByOrganization(_ context.Context, organizationID uuid.UUID) ([]database.ProvisionerDaemon, error) {
+func (q *FakeQuerier) GetProvisionerDaemonsByOrganization(_ context.Context, arg database.GetProvisionerDaemonsByOrganizationParams) ([]database.ProvisionerDaemon, error) {
 	q.mutex.RLock()
 	defer q.mutex.RUnlock()
 
 	daemons := make([]database.ProvisionerDaemon, 0)
 	for _, daemon := range q.provisionerDaemons {
-		if daemon.OrganizationID == organizationID {
-			daemon.Tags = maps.Clone(daemon.Tags)
-			daemons = append(daemons, daemon)
+		if daemon.OrganizationID != arg.OrganizationID {
+			continue
+		}
+		if arg.Tags != nil {
+			for k, v := range arg.Tags {
+				if t, found := daemon.Tags[k]; !found || t != v {
+					continue
+				}
```

Comment:
To me it doesn't look like this replicates the query condition for empty jobs. A tagged provisioner could pick this up if the job is `'{"scope": "organization", "owner": ""}'` and the provisioner is `'{"scope": "organization", "owner": "", "foo": "bar"}'`

Reply:
Correct, in the case of `{"scope": "organization", "owner": ""}` then the tags must match exactly.

Reply:
fixed